src/SlackConnector/SlackConnection.cs

```diff
@@ -63,20 +63,26 @@ public void Initialise(ConnectionInformation connectionInformation)
             ConnectedSince = DateTime.Now;
         }
 
-        private async Task ListenTo(InboundMessage inboundMessage)
+        private Task ListenTo(InboundMessage inboundMessage)
         {
-            if (inboundMessage?.MessageType != MessageType.Message)
-                return;
-            if (string.IsNullOrEmpty(inboundMessage.User))
-                return;
-            if (!string.IsNullOrEmpty(Self.Id) && inboundMessage.User == Self.Id)
-                return;
+            if (inboundMessage == null) Task.FromResult(false);
 
-            if (inboundMessage.Channel != null && !_connectedHubs.ContainsKey(inboundMessage.Channel))
+            switch (inboundMessage.MessageType)
```

Comment:
Instead of using a switch statement, could this be replace with a `type` -> `func` dictionary? Then you would just need to do something like `dictionary[typeof(inboundMessage)](inboundMessage)` etc